GVFS/GVFS.Common/NamedPipes/NamedPipeStreamReader.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace GVFS.Common.NamedPipes
+{
+    /// <summary>
+    /// Implements the NamedPipe protocol as described in NamedPipeServer.
+    /// </summary>
+    public class NamedPipeStreamReader
+    {
+        public const int BufferLength = 1024;

```

Comment:
I think the only external code that uses this is unit tests, so please make it private. In my opinion it's not a good idea to share constants between production and test code, because they will just end up agreeing with each other no matter what.

Reply:
I was trying to enforce wat that we test sending messages that are larger than the buffer size. If we completely hide this, then we could update the production code to use a buffer larger than what the test is using.

What if we instead expose an option for callers to control the buffer size. Then the tests could force a smaller buffer size and ensure it is testing the expected functionality...

Reply:
I wouldn't do either of those things. A unit test should be testing a _requirement_, not knowledge of an _implementation_. So the unit test should decide how big of a message is reasonable to send and test that limit, and the implementation is then responsible for handling that.